### PR TITLE
feat(Todolist): Unit test coverage

### PR DESCRIPTION
**Pull request by tarusin**

### CODE REVIEW

File: src/app/app.component.spec.ts [hunk 1]
```diff
@@ -1,27 +1,191 @@
...
+      {
+        title: 'Task4',
+        completed: false
+      }
+    ];
+    fixture = TestBed.createComponent(AppComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('was created', async(() => {
```
Comment (EvilAlexei) [inline src/app/app.component.spec.ts]: Not necessary test. If component not created all tests will fail

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -1,27 +1,191 @@
...
-  }));
-  it(`should have as title 'app'`, async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
-    const app = fixture.debugElement.componentInstance;
-    expect(app.title).toEqual('app');
-  }));
-  it('should render title in a h1 tag', async(() => {
-    const fixture = TestBed.createComponent(AppComponent);
+
+  it('should return array of checked todos', () => {
```
Comment (EvilAlexei) [inline src/app/app.component.spec.ts]: According to our company style guide try to avoid word `should`

File: src/app/app.component.spec.ts [hunk 3]
```diff
@@ -1,27 +1,191 @@
...
+    component.todos = tasks;
+
+    component.checkedTodos();
+
+    expect(component.checkedTodos().length).toEqual(3);
+  });
+
+  it('should return todos length equal checked todos length', () => {
```
Comment (EvilAlexei) [inline src/app/app.component.spec.ts]: Too complicated description

File: src/app/app.component.spec.ts [hunk 4]
```diff
@@ -1,27 +1,191 @@
...
+    component.todos = tasks;
+    const task = component.todos[2];
+    component.filter = 'completed';
+
+    component.isShow(task);
+
+    expect(component.isShow(task)).toBe(true);
+  });
+
+  it('should call console.log("This is wrong !")', () => {
```
Comment (EvilAlexei) [inline src/app/app.component.spec.ts]: Too complicated, text should be more human

File: src/app/app.component.spec.ts [hunk 5]
```diff
@@ -1,27 +1,191 @@
...
+
+    expect(component.activeTodosCount().length).toBe(1);
+  });
+
+  it('should remove completed todos', () => {
+    spyOn(component, 'delete').and.callThrough();
+    component.todos = tasks;
+
+    component.removeCompleted();
     fixture.detectChanges();
```
Comment (SergeyKuryatnick) [inline src/app/app.component.spec.ts]: why is this needed?

File: src/app/app.component.spec.ts [hunk 6]
```diff
@@ -1,27 +1,191 @@
...
+  it('should return todos length equal checked todos length', () => {
+    component.todos = tasks;
+
+    component.isAllChecked();
+
+    expect(component.isAllChecked()).toBe(false);
+  });
+
+  it('should return completed todos length', () => {
+    component.todos = tasks.filter((todo: Todo) => todo.completed === true);
```
Comment (SergeyKuryatnick) [inline src/app/app.component.spec.ts]: component.todos = tasks;
component.todos[3].completed = true;

File: src/app/app.component.spec.ts [hunk 7]
```diff
@@ -1,27 +1,191 @@
...
+      declarations: [ AppComponent ],
+      schemas: [ NO_ERRORS_SCHEMA ]
+    });
+
+    tasks = [
+      {
+        title: 'Task1',
+        completed: true
+      },
+      {
```
Comment (SergeyKuryatnick) [inline src/app/app.component.spec.ts]: add one more task with `completed = false`
